state_test.go

```diff
@@ -325,9 +325,8 @@ func TestStateUpdateReportStatus(t *testing.T) {
 		s, c = s.Handle(&ctx, sc)
 		assert.IsType(t, &UpdateStatusReportState{}, s)
 		assert.False(t, c)
-		assert.WithinDuration(t, time.Now(), now.Add(retry), time.Millisecond)
-		now = time.Now()
 	}
+	assert.WithinDuration(t, time.Now(), now.Add(time.Duration(int64(shouldTry)*int64(retry))), time.Millisecond*10)
```

Comment:
I would do something like 
`assert.WithinDuration(t, now, time.Now(), shouldTry * retry + time.Millisecond*10)`.

Reply:
Yes, perhaps that's clearer. Will do that.